Scripts/Python/ki/xKIChat.py

```diff
@@ -1245,33 +1245,44 @@ def RestartGame(self, params):
     ## Look around for exits and informational text.
     def LookAround(self, params):
 
-        # Find the nearby people.
-        playerList = self.chatMgr.GetPlayersInChatDistance(minPlayers=-1)
-        people = "nobody in particular"
-        if len(playerList) > 0:
-            people = ""
-            for player in playerList:
-                people += player.getPlayerName() + ", "
-            people = people[:-2]
-
         # Load the Age-specific text.
         ageInfo = PtGetAgeInfo()
         if ageInfo is None:
             return
+
         currentAge = ageInfo.getAgeFilename()
         see = ""
         exits = " North and West."
+        people = ""
+        peopleVerb = "is"
         if currentAge in kEasterEggs:
             see = kEasterEggs[currentAge]["see"]
+
             if not kEasterEggs[currentAge]["exits"]:
                 exits = "... well, there are no exits."
             else:
-                exits = " " + kEasterEggs[currentAge]["exits"]
+                exits = kEasterEggs[currentAge]["exits"]
+
             if "people" in kEasterEggs[currentAge]:
                 people = kEasterEggs[currentAge]["people"]
 
+        # Find the nearby people if kEasterEggs didn't define people text override for the Age.
+        if not people:
+            playerList = self.chatMgr.GetPlayersInChatDistance(minPlayers=-1)
+            playerListLen = len(playerList)
+            people = " nobody in particular."
+            peopleVerb = "are" if playerListLen > 1 else "is"
+
+            if playerListLen > 0:
+                people = " "
+                for idx, player in enumerate(playerList):
+                    # concatenate player names together with commas (using "and" before the last name)
+                    people += player.getPlayerName() + (", " if idx != playerListLen - 2 else ", and ")
+                people = people[:-2]
+                people += "."
+
         ## Display the info.
-        self.chatMgr.AddChatLine(None, "{}: {} Standing near you is {}. There are exits to the{}".format(GetAgeName(), see, people, exits), 0)
+        self.chatMgr.AddChatLine(None, "{}: {} Standing near you {}{} There are exits to the{}".format(GetAgeName(), see, peopleVerb, people, exits), 0)
```

Comment:
All the string concatting and manual formating is performance tanking and tedious. Let's use the magic of Python 3 to make this better. Note that I didn't test this suggestion, so hopefully I didn't bungle the formatting.

```suggestion
        # Find the nearby people if kEasterEggs didn't define people text override for the Age.
        if not people:
            playerList = self.chatMgr.GetPlayersInChatDistance(minPlayers=-1)
            playerListLen = len(playerList)
            peopleVerb = "are" if playerListLen > 1 else "is"

            if playerListLen == 0:
                people = "nobody in particular."
            else:
                people = ", ".join((f"{'and ' if idx + 1 == playerListLen else ''}{player.getPlayerName()}" for idx, player in enumerate(playerList)))

        ## Display the info.
        self.chatMgr.AddChatLine(None, f"{GetAgeName()}: {see} Standing near you {peopleVerb} {people}. There are exits to the{exits}", 0)
```

Reply:
I'll try this out and tweak anything that needs to be changed. Not certain the best way to see if this performs better than before, or shall we just assume .join and f-strings will almost certainly be faster?

Reply:
I think my latest commit properly integrates the use of join and f-strings while keeping the addition of leading space and ending period when creating the nearby player string. Leading and trailing punctuation/spacing are included in the "people" string to ensure Age creators can more fully customize their own "people" value (e.g., using "..." to start or "?" to end).